### ISSUE TITLE
(Feature Request): Like/Dislike percentage

### ISSUE THREAD

**Issue by FreakingChicken** [Anarios/return-youtube-dislike#144]
### Extension or Userscript?

Both

### Request or suggest a new feature!

Add a toggleable like/dislike percentage similar to how VidIQ does it that you can see on every video before you click it and on the video page. This will help with quickly judging a videos quality at a glance.

### Ways to implement this!

Use the likes and dislikes to calculate a percentage with 100% being everyone liked it and 0% being everyone disliked it and colorize the percentage based on how much it is.

90%-100% Green
80%-89% Yellow
<79% Red

### Can you work on this?

- [ ] Yes
- [X] No

**Comment by Robin5605** [comment]
Slight problem with this would be that on load of the YouTube home page, it would have to make many many API calls to the API to get the dislikes and calculate the ratio - and keep doing this every time a new video is loaded. This might be quite intensive on the API with thousands of people doing it at once.

**Comment by mattphelps-8451** [comment]
What browser extension do people normally use to see like/dislike percentage?

**Comment by Robin5605** [comment]
This is the only good one I've come across so far - so not sure.

**Comment by KendallDoesCoding** [comment]
> What browser extension do people normally use to see like/dislike percentage?

VIDIQ

**Comment by FreakingChicken** [comment]
VidIQ had this feature for a long time. They recently removed it due to dislikes being removed. This is what it looked like.
![image](https://user-images.githubusercontent.com/10645484/144515405-0db42eb1-1fb7-44ec-ab81-4b71c9b75476.png)

**Comment by FreakingChicken** [comment]
Also a solution to too many api calls would be to cache the results. The likes of the video will change a lot but the ratio won't change that much.

**Comment by mphelp** [comment]
@FreakingChicken caching client-side in the browser?

**Comment by mphelp** [comment]
I'm doing some experimenting this week on approximate speed. Definitely bottlenecked by requiring 1 API call per related video

**Comment by Robin5605** [comment]
I think that the initial load would be the most stressful - say when you first open YouTube. You'd have to get like/dislikes for around about 30-40 videos, and as you scroll you get 10ish videos per row. But yes, please do some testing and let us know how bad it is.

**Comment by Robin5605** [comment]
> I'm doing some experimenting this week on approximate speed. Biggest bottleneck is requiring 1 API call per related video. I wonder if @Anarios has plans to expand API to include requests for multiple IDs at once

Yeah, I wish we had code for the API. It would help a lot with this + a bunch of issues. We could contribute code so that it takes bulk IDs and returns them.

**Comment by KendallDoesCoding** [comment]
> VidIQ had this feature for a long time. They recently removed it due to dislikes being removed. This is what it looked like. ![image](https://user-images.githubusercontent.com/10645484/144515405-0db42eb1-1fb7-44ec-ab81-4b71c9b75476.png)

Yeah, even I commented on this with VIDIQ but deleted the comment, has they deketed the feature.

**Comment by RicksNext** [comment]
> I'm doing some experimenting this week on approximate speed. Biggest bottleneck is requiring 1 API call per related video. I wonder if @Anarios has plans to expand API to include requests for multiple IDs at once

The YouTube API allows you to get data from many videos (idk the limit but my guess would be 100 videos at once) in a single request, the only problem would be the YT Quota maybe running out quicker, but that would allow the system to archive even more videos on YouTube at a massive rate.

**Comment by GoulartNogueira** [comment]
Let's split the problems apart:
1. Show Like/Dislike percentage
2. Show some Dislike information on multiple videos while navigating
    - Only this one requires a lot of API calls.

### I suggest we focus on problem 1:
When the user opens a video and we already have the dislike data, we we can just make a simple math division and show it with CSS.

### Why do I need that?
Sometimes it's hard to estimate a score in a glance. So dislikes are just a meaningless number.

Suppose I have 3 videos about a topic. Which is a better source of information?
| | | |
| - | - | - |
| 👍 97 | 👎 10 | ~10% |
| 👍 37.4M | 👎 578,318 | **??? %** |
| 👍 7.4M | 👎 22,150 | **??? %** |

On the first example, it's easy to notice the video is probably bad, with +10% dislikes.
What about the other examples?
If I had a percentage, it would become much easier to compare them:

| | | |
| - | - | - |
| 👍 97 | 👎 10 | $${\color{red} 10 \\% dislike }$$ |
| 👍 37.4M | 👎 578,318 | $${\color{green} 1.5 \\% dislike }$$ |
| 👍 7.4M | 👎 22,150 | $${\color{darkOrange} 3 \\% dislike }$$ |

**Comment by GoulartNogueira** [comment]
I've just realized there were some hidden setting and this (partially) implemented https://github.com/Anarios/return-youtube-dislike/issues/48

Also, check this other extension to handle that:
https://github.com/elliotwaite/thumbnail-rating-bar-for-youtube

Please consider closing this issue
